File: python/githubauthentication/github_device_code_response.py
class GitHubDeviceCodeResponse:
	def __init__(
			  self,
			  device_code: str,
			  user_code: str,
			  verification_uri: str,
			  expires_in: int,
			  interval: int) -> None:
		self._device_code = device_code
		self._user_code = user_code
		self._verification_uri = verification_uri
		self._expires_in = expires_in
		self._interval = interval
	@property
	def device_code(self):
		return self._device_code
	
	@property
	def user_code(self):
		return self._user_code
	@property
	def verification_uri(self):
		return self._verification_uri
	@property
	def expires_in(self):
		return self._expires_in
	@property
	def interval(self):
		return self._interval
	

File: php/src/GitHubAuthentication/GitHubDeviceCodeResponse.php
<?php

namespace Baywet\Githubtododemo\Githubauthentication;

class GitHubDeviceCodeResponse 
{
	public string $device_code;
	public string $user_code;
	public string $verification_uri;
	public int $expires_in_seconds;
	public int $interval_in_seconds;
	public static function fromJson($jsonPayload)
	{
		$jsonPayload = json_decode($jsonPayload, true);
		$instance = new GitHubDeviceCodeResponse();
		if (array_key_exists("device_code", $jsonPayload)) {
			$instance->device_code = $jsonPayload["device_code"];
		}
		if (array_key_exists("user_code", $jsonPayload)) {
			$instance->user_code = $jsonPayload["user_code"];
		}
		if (array_key_exists("verification_uri", $jsonPayload)) {
			$instance->verification_uri = $jsonPayload["verification_uri"];
		}
		if (array_key_exists("expires_in", $jsonPayload)) {
			$instance->expires_in_seconds = $jsonPayload["expires_in"];
		}
		if (array_key_exists("interval", $jsonPayload)) {
			$instance->interval_in_seconds =$jsonPayload["interval"];
		}
		return $instance;
	}
}